### ISSUE TITLE
Playwright tests fail in Safari when using Chakra with NextJS

### ISSUE THREAD

**Issue by piersolenski** [chakra-ui/chakra-ui#8130]
### Description

I was trying to setup tests for my Next JS + Chakra UI project, however they kept failing in Safari as the page would load with the "Application error: a client-side exception has occurred (see the browser console for more information)." error. At first I thought this was just my local setup, however if you run the same test on the https://chakra-ui.com/ website you will get the same error.

```js
test("should navigate to the Installation page", async ({ page }) => {
  await page.goto("https://chakra-ui.com");
  await page.click('[href="/getting-started"]');
  await expect(page).toHaveURL("https://chakra-ui.com/getting-started");
  await expect(page.getByRole("heading", { level: 1 })).toContainText(
    "Installation",
  );
});
```

<img width="817" alt="CleanShot 2023-11-22 at 16 00 30@2x" src="https://github.com/chakra-ui/chakra-ui/assets/1285419/69b0c6a3-16e1-4eda-bcec-baf984d3a467">

It passes fine in other browsers.

### Link to Reproduction

https://github.com/piersolenski/chakra-playwright

### Steps to reproduce

1. `git clone git@github.com:piersolenski/chakra-playwright.git`
2. `npm i`
3. `npm run dev` / `npm run build && npm start`
4. `npx playwright test --debug`

### Chakra UI Version

2.2.0

### Browser

Safari

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

Playwright config:

```js
import { defineConfig, devices } from "@playwright/test";
import path from "path";

// Use process.env.PORT by default and fallback to port 3000
const PORT = process.env.PORT || 3000;

// Set webServer.url and use.baseURL with the location of the WebServer respecting the correct set port
const baseURL = `http://localhost:${PORT}`;

// Reference: https://playwright.dev/docs/test-configuration
export default defineConfig({
  // Timeout per test
  timeout: 30 * 1000,
  // Test directory
  testDir: path.join(__dirname, "e2e"),
  // If a test fails, retry it additional 2 times
  retries: 1,
  // Artifacts folder where screenshots, videos, and traces are stored.
  outputDir: "test-results/",

  // Run your local dev server before starting the tests:
  // https://playwright.dev/docs/test-advanced#launching-a-development-web-server-during-the-tests
  webServer: {
    command: "npm run dev",
    url: baseURL,
    timeout: 120 * 1000,
    reuseExistingServer: !process.env.CI,
  },

  use: {
    // Use baseURL so to make navigations relative.
    // More information: https://playwright.dev/docs/api/class-testoptions#test-options-base-url
    baseURL,

    // Retry a test if its failing with enabled tracing. This allows you to analyze the DOM, console logs, network traffic etc.
    // More information: https://playwright.dev/docs/trace-viewer
    trace: "retry-with-trace",

    // All available context options: https://playwright.dev/docs/api/class-browser#browser-new-context
    // contextOptions: {
    //   ignoreHTTPSErrors: true,
    // },
  },

  projects: [
    // {
    //   name: "Desktop Chrome",
    //   use: { ...devices["Desktop Chrome"] },
    // },
    {
      name: "webkit",
      use: { ...devices["Desktop Safari"] },
    },
  ],
});
```

**Comment by moeriki** [comment]
Fixed in STP Release 184 (Safari 17.4, WebKit 19618.1.7.1.1)

**Comment by segunadebayo** [comment]
Closing based on the https://github.com/chakra-ui/chakra-ui/issues/8130#issuecomment-1847041714
